Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceToTemplatePropagator.h

```diff
@@ -33,7 +33,7 @@ namespace AzToolsFramework
 
             InstanceOptionalReference GetTopMostInstanceInHierarchy(AZ::EntityId entityId) override;
 
-            bool PatchTemplate(PrefabDomValue& providedPatch, TemplateId templateId, InstanceOptionalReference instanceToExclude = AZStd::nullopt) override;
+            bool PatchTemplate(PrefabDomValue& providedPatch, TemplateId templateId, InstanceOptionalReference instanceToExclude = AZStd::nullopt, bool immediate = false) override;
```

Comment:
Since instanceToExclude will likely be removed down the line, but immediate will likely stay, I'd recommend switching their place in the signature to simplify future changes.
That should also simplify some calls later on (no need to pass nullopt)

Reply:
It might be worth (depending on the number of changes required) to update this parameter to be a `struct` that contains both parameters (and potentially more in future). This makes future changes easier (and we don't end up with `SomeFunction(true, false, false, true)`... down the line 😉). It can make the call sites clearer too. Also another improvement would be to use an `enum` as opposed to a `bool` (say `PatchType::Immediate` and `PatchType::Deferred`). This also makes the calling code much easier to understand